vowpalwabbit/debug_print.h

```diff
@@ -24,4 +26,22 @@ std::ostream& operator<<(std::ostream& os, const std::vector<T>& vec)
   os << " ]";
   return os;
 }
-}  // namespace std
\ No newline at end of file
+}  // namespace std
+
+template <class T>
```

Comment:
nit: Personally I like to keep object-specific functionality with the object (that way I can just include `v_array.h` when I want to print one instead of needing both that and `debug_print.h`)

Reply:
That requires iostream to be in the v_array header which is relatively heavy. That's the motivating factor here